node_modules/oae-tenants/lib/api.js

```diff
@@ -203,6 +230,55 @@ var _cacheTenants = function(callback) {
             tenantsByHost[tenant.host] = tenant;
         });
 
+        // Cache the sorted list
+        sortedTenants = _.sortBy(tenants, 'alias');
+
+        // Indicate that all tenants have been cached
+        TenantsAPI.emit('cached');
+
+        return callback(null, tenants);
+    });
+};
+
+/**
+ * Fetch the tenant and update it's entry in the cache
+ *
+ * @param  {String}         tenantAlias         The alias of the tenant to be re-cached
+ * @param  {Function}       [callback]          Standard callback function
+ * @param  {Object}         [callback.err]      Error object containing the error code and message
+ * @api private
+ */
+var _updateCachedTenant = function(tenantAlias, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err}, 'Failed to re-cache the tenants');
+        }
+    };
+
+    // Get the available tenants
+    Cassandra.runQuery('SELECT * FROM "Tenant" WHERE "alias" = ?', [tenantAlias], function(err, rows) {
+        if (err) {
+            return callback(err);
+        }
+
+        _.each(rows, function(row) {
+            var tenant = mapToTenant(row);
+            // Cache all tenants
+            tenants[tenant.alias] = tenant;
+            tenantsByHost[tenant.host] = tenant;
+
+            // Insert at the correct location in the sorted list
+            var index = _.findIndex(sortedTenants, function(t){
+                return t.alias === tenantAlias;
+            });
+            if (index === -1) {
+                index = _.sortedIndex(sortedTenants, tenant, 'alias');
+                sortedTenants.splice(index, 0, tenant);
+            } else {
+                sortedTenants[index] = tenant;
+            }
+        });
```

Comment:
There will only be 1 row here, so representing as a foreach is misleading. It should just grab the first one, and warn if there aren't any